src/js/modal-dialog.js

```diff
@@ -0,0 +1,386 @@
+/**
+ * @file modal-dialog.js
+ */
+import document from 'global/document';
+import tsml from 'tsml';
+
+import * as Dom from './utils/dom';
+import * as Events from './utils/events';
+import * as Fn from './utils/fn';
+import log from './utils/log';
+
+import Component from './component';
+import CloseButton from './close-button';
+
+const CLASS_NAME_PREFIX = 'vjs-modal-dialog';
+
+// An array of strings which are used for modal sub-element classes,
+// so will cause CSS issues if used as a slug.
+const DISALLOWED_SLUGS = [
+  'content'
+];
+
+const ESC = 27;
+
+/**
+ * The `ModalDialog` displays over the video and its controls, which blocks
+ * interaction with the player until it is closed.
+ *
+ * Modal dialogs include a "Close" button and will close when that button
+ * is activated - or when ESC is pressed anywhere.
+ *
+ * @extends Component
+ * @class ModalDialog
+ */
+class ModalDialog extends Component {
+
+  /**
+   * Constructor for modals.
+   *
+   * @param  {Player} player
+   * @param  {Object} [options]
+   * @param  {Mixed} [options.content=undefined]
+   *         Provide customized content for this modal.
+   *
+   * @param  {Boolean} [options.disposeOnClose=false]
+   *         If `true`, the modal can only be opened once. It will be
+   *         disposed as soon as it's closed, which is useful for one-off
+   *         modals.
+   *
+   * @param  {Boolean} [options.fillAlways=false]
```

Comment:
Another candidate for change would be to rename the `open`/`close` methods to `show`/`hide` (overriding/expanding on those methods of `Component`).